router/mocks_test.go

```diff
@@ -0,0 +1,113 @@
+// No mocks are tested by this file.
+// It supplies some mock implementations to other unit tests,
+// and is named "...test.go" so it is only compiled under `go test`.
+
+package router
+
+import (
+	wt "github.com/zettio/weave/testing"
+	"net"
+	"testing"
+)
+
+type peerQueue struct {
+	peers []*Peer
+}
+
+func (q *peerQueue) clear() {
+	q.peers = nil
+}
+
+// Construct a Router object with a mock interface to check peer
+// garbage-collection, and without firing up any ancilliary goroutines
+func NewTestRouter(t *testing.T, name PeerName, queue *peerQueue) *Router {
+	onMacExpiry := func(mac net.HardwareAddr, peer *Peer) {}
+	onPeerGC := func(peer *Peer) {
+		//t.Log("Removing unreachable", peer)
+		if queue != nil {
+			queue.peers = append(queue.peers, peer)
+		}
+	}
```

Comment:
Instead of instrumenting the gc with a callback that collects gc'ed peers, a better test would to be check whether Peers contain exactly the peers we expect.